tests/phpunit/tests/admin/includesMedia.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+/**
+ * @group media
+ * @group admin
+ */
+class Tests_Admin_includesMedia extends WP_UnitTestCase {
+	/** @var \WP_Post */
+	protected static $post;
+	/** @var \WP_Post */
+	protected static $different_post;
+
+	public static function wpSetUpBeforeClass( \WP_UnitTest_Factory $factory ) {
+		self::$post           = $factory->post->create_and_get();
+		self::$different_post = $factory->post->create_and_get();
+	}
+
+	public function wpTearDownAfterClass() {
+		self::$post           = null;
+		self::$different_post = null;
+	}
+
+	public function setUp() {
+		parent::setUp();
+
+		add_filter( 'pre_http_request', [ $this, 'internal_fake_download_url' ], 10, 3 );
+	}
+
+	public function tearDown() {
+		parent::tearDown();
+
+		$this->remove_added_uploads();
+		remove_filter( 'pre_http_request', [ $this, 'internal_fake_download_url' ] );
+	}
+
+	/**
+	 * Filters whether to preempt an HTTP request's return value. Mocking the
+	 * response. Since ticket 49631
+	 *
+	 * @param false  $preempt Whether to preempt an HTTP request's return value. Default false.
+	 * @param array  $parsed_args HTTP request arguments.
+	 * @param string $url The request URL.
+	 *
+	 * @return array
+	 */
+	public function internal_fake_download_url( $preempt, array $parsed_args, $url = '' ) {
+		// Just need an image content string to fill the new image, keeping the same mime type.
+		file_put_contents( $parsed_args['filename'], file_get_contents( DIR_TESTDATA . '/images/canola.jpg' ) );
+
+		return [
+			'response' => [
+				'code' => 200,
+			],
+		];
+	}
+
+	/**
+	 * @ticket 49631
+	 */
+	public function test_media_sideload_image() {
```

Comment:
This should be broken up into several separate test methods—one for each assertion being made—or to refactor this to work with a data provider so you can run multiple independent tests by passing different inputs to a single test method.

Also, through each of these assertions, the "expected" value should be passed as the first parameter and the value being tested should be second.

Reply:
I agree this could be broken in several tests. I thought was for the same purpose, test the return value of the function and the meta of the post attached.

I'm pushing a change with the latest assert that I mix the order 👍